### ISSUE TITLE
Support searching from Muti JSON documents ?

### ISSUE THREAD

**Issue by icewukong** [tidwall/tile38#142]
For example:
```
   SET roads road1 OBJECT {"type": "LineString",coordinates": [[102.0, 0.0], [103.0, 1.0], [104.0,0.0],[105.0, 1.0]]},"properties": {"prop0": "value0","prop1": 0.0}}
   SET roads road2 OBJECT {"type": "LineString",coordinates": [[102.0, 0.0], [103.0, 1.0], [104.0,0.0],[105.0, 1.0]]},"properties": {"prop0": "value1","prop1": 2.0}}
```
I want to return the object that it's "prop0" equal "value1",with a command:
`JSEARCH key MATCH path`
e.g.
`JSEARCH roads MATCH properties.#[prop0=="value1"]#`

**Comment by tidwall** [comment]
It's possible, but it would be slow. An O(n) operation. Would that be OK?

**Comment by icewukong** [comment]
How about to create a index,like buntdb?
Despite this will expose the concept of the index.

**Comment by icewukong** [comment]
BTW,
Is the SEARCH operation on type STRING  a O(n) operation?

**Comment by tidwall** [comment]
No, the STRING type goes into a Btree. It's in binary order. O(log n)
We may be able to add a command for creating indexes for collections. Similar to [SETINDEX](https://github.com/tidwall/summitdb/wiki/SETINDEX) in SummitDB.

**Comment by icewukong** [comment]
That would be best！

**Comment by tidwall** [comment]
All right. I'll investigate further. Thanks for you input.

**Comment by icewukong** [comment]
Will the ability that searching from Muti JSON documents  be available in version 2.x ?

**Comment by tidwall** [comment]
I hope so. I'm experimenting with new SETINDEX & DELINDEX commands, and an updated SEARCH command that will allow for indexed queries on the STRING type.

**Comment by icewukong** [comment]
Looking forward to your project 2.0  :+1:

**Comment by kirantpatil** [comment]
Hi,

May I know the commands to search with json field using NEARBY and WHERE ?

For example:

> 
> 127.0.0.1:9851> jset test 1 outstation dharwad
> {"ok":true,"elapsed":"49.379µs"}
> 
> 127.0.0.1:9851> get test 1
> {"ok":true,"object":{"type":"Point","coordinates":[-84.4767280575,45.6304059969,100],"outstation":"dharwad"},"elapsed":"34.099µs"}
> 
> 127.0.0.1:9851> nearby test where outstation dharwad 6000
> (error) invalid argument 'dharwad'
>

**Comment by kirantpatil** [comment]
> I hope so. I'm experimenting with new SETINDEX & DELINDEX commands, and an updated SEARCH command that will allow for indexed queries on the STRING type.

@tidwall: May I know any updates on this feature ?

**Comment by tidwall** [comment]
Hi @kirantpatil. Sorry but this feature is on hold and I don’t have any plans at the moment to implement right now.
